### ISSUE TITLE
AD0001 - NullReferenceException when running CA2246

### ISSUE THREAD

**Issue by RobSiklos** [dotnet/roslyn-analyzers#2889]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

[v2.9.5](https://github.com/dotnet/roslyn-analyzers/releases/tag/v2.9.5)

#### Diagnostic ID

[CA2246](https://github.com/dotnet/roslyn-analyzers/pull/2717)

#### Repro steps

Analyze the following code:
```csharp
public static class Class1
{
    public static void Foo()
    {
        var u = new System.UriBuilder();
        u.Host = u.Path = string.Empty;
    }
}
```
#### Expected behavior
No warning or error.

#### Actual behavior

> Analyzer 'Microsoft.CodeQuality.Analyzers.QualityGuidelines.AssigningSymbolAndItsMemberInSameStatement' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'
> 
> Analyzer 'Microsoft.CodeQuality.Analyzers.QualityGuidelines.AssigningSymbolAndItsMemberInSameStatement' threw the following exception:
> 'Exception occurred with following context:
> Compilation: RoslynAnalyzersFalsePositives
> IOperation: SimpleAssignment
> SyntaxTree: D:\src\Rob\RoslynAnalyzersFalsePositives\RoslynAnalyzersFalsePositives\CA2246.cs
> SyntaxNode: u.Host = u.Path = string.Empty [AssignmentExpressionSyntax]@[193..223) (7,12)-(7,42)
> 
> System.NullReferenceException: Object reference not set to an instance of an object.
>    at Microsoft.CodeQuality.Analyzers.QualityGuidelines.AssigningSymbolAndItsMemberInSameStatement.<>c.<AnalyzeAssignment>b__8_0(ILocalReferenceOperation a, ILocalReferenceOperation b)
>    at Microsoft.CodeQuality.Analyzers.QualityGuidelines.AssigningSymbolAndItsMemberInSameStatement.AnalyzeAssignmentToMember[T](ISimpleAssignmentOperation assignmentOperation, T instance, Func`3 equalityComparer)
>    at Microsoft.CodeQuality.Analyzers.QualityGuidelines.AssigningSymbolAndItsMemberInSameStatement.AnalyzeAssignment(OperationAnalysisContext context)
>    at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c.<ExecuteOperationAction>b__54_0(ValueTuple`2 data)
>    at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)

**Comment by mavasani** [comment]
Thanks for the prompt bug report @RobSiklos. I will work on a fix and point you a myget package with the fix. Based on number of folks that hit this, we may have to release a 2.9.6-beta1 or 2.9.6 for this fix.

**Comment by mavasani** [comment]
@RobSiklos @abelykh0 can you please try https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.6-beta1.19502.2+ce473561?

**Comment by RobSiklos** [comment]
Confirmed fixed in 2.9.6-beta1.19502.2+ce473561. Thanks @mavasani !

**Comment by mavasani** [comment]
FYI: We got large number of customer reports for this AD0001. So, we have just published 2.9.6 with this fix: https://github.com/dotnet/roslyn-analyzers/releases/tag/v2.9.6

Thanks everyone for helping find and fix this issue promptly.

**Comment by adamralph** [comment]
I'm [using 2.10.0](https://github.com/Particular/NServiceBus/blob/d39fa22d4894f87fe3b8c6d6e0569c55770f8946/src/NServiceBus.Core.Analyzer/NServiceBus.Core.Analyzer.csproj#L14) and I'm [still having this problem](https://github.com/Particular/NServiceBus/pull/6051/checks?check_run_id=2518127665).

**Comment by adamralph** [comment]
Sorry, it's a different error message. I guess I'll raise a new issue...

**Comment by adamralph** [comment]
https://github.com/dotnet/roslyn-analyzers/issues/1803#issuecomment-833493379
